out/blocks/wd_admin_payment_main_form.tpl

```diff
@@ -2,31 +2,93 @@
     .pb-35 {
         padding-bottom: 35px;
     }
+
+    .color-success {
+      color: green;
+    }
+
+    .color-failure {
+      color: red;
+    }
 </style>
 
 <script type="text/javascript">
+  function addClassToElements(elementIds, className) {
+    elementIds.forEach(function(elementId) {
+      var element = document.getElementById(elementId);
+
+      if (element) {
+        element.classList.add(className);
+      }
+    });
+  }
+
+  function removeClassFromElements(elementIds, className) {
+    elementIds.forEach(function(elementId) {
+      var element = document.getElementById(elementId);
+
+      if (element) {
+        // if no class name was passed, remove all classes from the element
+        if (!className) {
+          element.className = '';
+        } else {
+          element.classList.remove(className);
+        }
+      }
+    });
+  }
+
   function testPaymentMethodCredentials() {
-    // the DOM element the test result will be rendered to
-    var resultSpan = document.getElementById('test_credentials_result');
+    var checkSuccess = false;
+
+    var successText = '[{oxmultilang ident="success_credentials"}]';
+    var failureText = '[{oxmultilang ident="error_credentials"}]';
+
+    var successClassName = 'color-success';
+    var failureClassName = 'color-failure';
+
+    // the id of the DOM element the test result will be rendered to
+    var resultSpanId = 'test_credentials_result';
+
+    // the ids of the DOM elements that should be marked red upon test failure
+    var labelConfigApiUrlId = 'labelConfigApiUrl';
```

Comment:
It's better to query the DOM at this point and store the elements instead of their IDs.

Reply:
That's indeed better, thanks!